coreference/bert_stuff_new.py

```diff
@@ -0,0 +1,202 @@
+
+    Returns
+    -------
+        arr of vectors
+    """
+    parallel_model.eval()
+
+    # Store vectors here
+    all_vectors = []
+    
+    
+    
+    #vector_map_path = working_folder + f'/{men_type}_vectormaps.pkl'
+
+    loc_mentions = []
+
+
+
+
+    batch = [bert_sentences[x] for x in range( len(bert_sentences))]
+    bert_tokens = parallel_model.module.tokenizer(batch, pad_to_max_length=True, add_special_tokens=False, truncation =False )
+    #print(batch)
+    input_ids = torch.tensor(bert_tokens['input_ids'], device=device)
+    attention_mask = torch.tensor(bert_tokens['attention_mask'], device=device)
+    #print(input_ids.shape)
+
+    m = input_ids.cpu()
+    k = m == parallel_model.module.vals[0]
+    p = m == parallel_model.module.vals[1]
+
+    v = (k.int() + p.int()).bool()
+    nz_indexes = v.nonzero()[:, 1].reshape(m.shape[0], 2)
+    loc_mentions.extend(nz_indexes.cpu().numpy())
+
+    #get the overflowing token indices and chunk the document so that the special tokens are within max token length(4096)
+    overflow_idx=[]
+  
+    for i,j in enumerate(loc_mentions):
+        if j[1]>=4096:
+            overflow_idx.append([i, j[1] ])
+   
+
+
+
+    idx = [x[0] for x in overflow_idx]
+    m_new  = []
+    for i, j in enumerate(overflow_idx):
+        bert_sent = bert_sentences[j[0] ]
+        m_new.append(m[j[0],  j[1]-4096+1:j[1]+1] )
+
+    m_new_tensor = torch.stack(m_new) 
+    m_new_tensor = F.pad(m_new_tensor,pad =(0, m.shape[1]-m_new_tensor.shape[1]), value=1) 
+
+    m[idx] = m_new_tensor
+
+    k = m == parallel_model.module.vals[0]
+    p = m == parallel_model.module.vals[1]
+
+    v = (k.int() + p.int()).bool()
+    nz_indexes = v.nonzero()[:, 1].reshape(m.shape[0], 2)
+
+    q = torch.arange(m.shape[1])
+    q = q.repeat(m.shape[0], 1)
+
+    msk_0 = (nz_indexes[:, 0].repeat(m.shape[1], 1).transpose(0, 1)) <= q
+    msk_1 = (nz_indexes[:, 1].repeat(m.shape[1], 1).transpose(0, 1)) >= q
+
+    msk_0_ar = (nz_indexes[:, 0].repeat(m.shape[1], 1).transpose(0, 1)) < q
+    msk_1_ar = (nz_indexes[:, 1].repeat(m.shape[1], 1).transpose(0, 1)) > q
+
+    attention_mask_g = msk_0.int() * msk_1.int()
+
+    input_ids = input_ids[:, :4096]
+    # attention_mask = attention_mask[:, :4096]
+    attention_mask[:, 0] = 2
+    attention_mask[attention_mask_g == 1] = 2
+    attention_mask = attention_mask[:, :4096]
+    arg1 = msk_0_ar.int() * msk_1_ar.int()
+    arg1 = arg1[:, :4096]
+    arg1 = arg1.to(device)
+
+
+    with torch.no_grad():
+        for i in tqdm(range(0, len(bert_sentences), batch_size), desc="generating "):
+
+            men_vectors = parallel_model(input_ids[i:i+batch_size], attention_mask[i:i+batch_size], arg1[i:i+batch_size])
+            all_vectors.extend(men_vectors)
+    
+            
+            
+            
+            
+            
+
+    return all_vectors
+
+
+def generate_cdlm_embeddings(mention_map, vec_map_path , key_name ='bert_doc', num_gpus=4, batch_size=150, cpu=False):
```

Comment:
clean this entire function. there are too many spaces... unnecessary commented code lines. unnecessary spaces.. always try to have no more than 1 line space between lines in a function. try adding documentation to the blocks of codes doing specific thing